### PR TITLE
add logic to handle break change scenarios for restructured java library 

### PR DESCRIPTION
**Pull request by kaibocai**
### Issue describing the changes in this PR

add logic to handle break change when customer using old java library and maven plugin where worker using restructured library. On old plugin the java library jar will not be put in customer lib folder. Worker now only depends on core library so it means worker cannot get annotation class from customer code. Add logics to handle this and avoid the break change. 

### Pull request checklist

* [ ] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [ ] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [ ] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

**Review by shreyas-gopalakrishna — COMMENTED** [review]
Let us update to lib_worker/azure-functions-java-library-2.0.0-SNAPSHOT.jar at all places

**Review by shreyas-gopalakrishna — DISMISSED** [review]

File: src/main/java/com/microsoft/azure/functions/worker/broker/JavaFunctionBroker.java [hunk 1]
```diff
@@ -94,17 +94,64 @@ private void addSearchPathsToClassLoader(FunctionMethodDescriptor function) thro
...
+		}
+	}
+
+	public void addJavaAnnotationLibrary() throws IOException {
+		String javaLibPath = System.getenv(Constants.FUNCTIONS_WORKER_DIRECTORY) + Constants.JAVA_LIBRARY_DIRECTORY;
+		File javaLib = new File(javaLibPath);
+		File[] files = javaLib.listFiles(file -> file.getName().contains(Constants.JAVA_LIBRARY_ARTIFACT_ID));
+		if (files.length == 0) throw new FileNotFoundException("Error loading java annotation library jar, no jar find from path:  " + javaLibPath);
+		if (files.length > 1) throw new FileNotFoundException("Error loading java annotation library jar, multiple jars find from path:  " + javaLibPath);
+		classLoaderProvider.addUrl(files[0].toURI().toURL());
```
Comment (amamounelsayed) [inline src/main/java/com/microsoft/azure/functions/worker/broker/JavaFunctionBroker.java:145]: Based on Fabio feedback can we add warning in this case?

Comment (kaibocai) [reply src/main/java/com/microsoft/azure/functions/worker/broker/JavaFunctionBroker.java:145]: added log info, please check.

File: package-pipeline.ps1 [hunk 2]
```diff
@@ -23,6 +23,7 @@ Get-ChildItem -Path .\target\* -Include 'azure*' -Exclude '*shaded.jar','*tests.
 StopOnFailedExecution
 copy-item ./worker.config.json pkg
 copy-item ./tools/AzureFunctionsJavaWorker.nuspec pkg/
+Copy-Item ".\annotationLib" pkg\annotationLib -Recurse
```
Comment (amamounelsayed) [inline package-pipeline.ps1]: Can we please add / instead of \?

Comment (kaibocai) [reply package-pipeline.ps1]: I am following v3 style on this line, but I can change it to / for sure

Comment (amamounelsayed) [reply package-pipeline.ps1]: Thank you Kaibo, yes please let's have it as /

**Review by amamounelsayed — APPROVED** [review]

**Pull request merged**